sunpy/net/dataretriever/sources/noaa.py

```diff
@@ -244,64 +205,18 @@ def fetch(self, qres, path=None, error_callback=None, **kwargs):
         -------
         Results Object
         """
-        urls = [qrblock['url'] for qrblock in qres]
-        filenames = []
-        local_filenames = []
-        for url, qre in zip(urls, qres):
-            name = url.split('/')[-1]
-            day = qre['Start Time']
-            if name not in filenames:
-                filenames.append(name)
-            if name.endswith('.gz'):
-                local_filenames.append('{}SRS.txt'.format(day.strftime('%Y%m%d')))
-            else:
-                local_filenames.append(name)
-
         if path is not None:
             path = pathlib.Path(path)
+        # Need to login to the FTP server.
+        urls = [qrblock['url'].replace("ftp://ftp", "ftp://anonymous:anon@@ftp")
```

Comment:
Adding login info to the url that is passed into scraper breaks it.
So I add it here so parfive picks it up via urlparse.

Reply:
I don't think I actually need to do this.